File: core/src/main/java/pcb/uwu/units/composite/electromagnetism/ElectricResistanceUnit.java
package pcb.uwu.units.composite.electromagnetism;

import pcb.uwu.core.CompositeUnit;
import pcb.uwu.core.UnitCounter;
import pcb.uwu.units.quantity.ElectricCurrentUnit;
import pcb.uwu.units.quantity.LengthUnit;
import pcb.uwu.units.quantity.MassUnit;
import pcb.uwu.units.quantity.TimeUnit;

public class ElectricResistanceUnit extends CompositeUnit {

	// region base constructor

	public ElectricResistanceUnit(MassUnit massUnit, LengthUnit lengthUnit, ElectricCurrentUnit electricCurrentUnit, TimeUnit timeUnit) {
		super(new UnitCounter()
				.major(massUnit)
				.major(lengthUnit, 2)
				.minor(timeUnit, 3)
				.minor(electricCurrentUnit, 2));
	}

	// endregion

	// region composite constructors

	public ElectricResistanceUnit(ElectricConductanceUnit electricConductanceUnit) {
		super(new UnitCounter()
				.minor(electricConductanceUnit.getUnitCounter()));
	}

	// endregion

}

File: dimension/src/main/java/pcb/uwu/unit/derived/electromagnetism/ElectricResistanceUnit.kt
package pcb.uwu.unit.derived.electromagnetism

import pcb.uwu.core.CompositeUnit
import pcb.uwu.core.UnitCounter
import pcb.uwu.unit.base.ElectricCurrentUnit
import pcb.uwu.unit.base.LengthUnit
import pcb.uwu.unit.base.MassUnit
import pcb.uwu.unit.base.TimeUnit

open class ElectricResistanceUnit : CompositeUnit
{
    constructor(massUnit: MassUnit,
                lengthUnit: LengthUnit,
                electricCurrentUnit: ElectricCurrentUnit,
                timeUnit: TimeUnit)
            : super(UnitCounter()
                        .major(massUnit)
                        .major(lengthUnit)
                        .major(lengthUnit)
                        .minor(timeUnit)
                        .minor(timeUnit)
                        .minor(timeUnit)
                        .minor(electricCurrentUnit)
                        .minor(electricCurrentUnit))

    constructor(electricConductanceUnit: ElectricConductanceUnit)
            : super(UnitCounter()
                        .minor(electricConductanceUnit))

    constructor(electricPotentialUnit: ElectricPotentialUnit,
                electricCurrentUnit: ElectricCurrentUnit)
            : super(UnitCounter()
                        .major(electricPotentialUnit)
                        .major(electricCurrentUnit))
}